wolf-test-utils/test/Wolf/Data/Export/TypesSpec.hs

```diff
@@ -15,3 +15,6 @@ spec = do
     eqSpec @Repo
     genValidSpec @Repo
     jsonSpecOnValid @Repo
+    functorSpecOnValid @ResultS
+    applicativeSpecOnValid @ResultS
+    monadSpecOnValid @ResultS
```

Comment:
COOL HUH!

Reply:
Yes!

Reply:
Now you need to implement it such that this test is automatically generated, and that in addition he asks me "do you want to have an Eq instance for Result w e a? If so, here is my suggestion: [..]", and that I can just press 'y' and bam, default implementation. As in, a system which can
1. figure out that there is a default implementation for Eq and Show for this new type
2. can generate the code which generates the default implementation
3. automatically suggests it